samples/multiplatform/kmp-android-app/src/main/kotlin/com/apollographql/apollo/kmpsample/data/ApolloCoroutinesService.kt

```diff
@@ -0,0 +1,49 @@
+package com.apollographql.apollo.kmpsample.data
+
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.coroutines.toDeferred
+import com.apollographql.apollo.kmpsample.fragment.RepositoryFragment
+import com.apollographql.apollo.kmpsample.GithubRepositoriesQuery
+import com.apollographql.apollo.kmpsample.GithubRepositoryCommitsQuery
+import com.apollographql.apollo.kmpsample.GithubRepositoryDetailQuery
+import com.apollographql.apollo.kmpsample.fragment.RepositoryDetail
+import com.apollographql.apollo.kmpsample.type.OrderDirection
+import com.apollographql.apollo.kmpsample.type.PullRequestState
+import com.apollographql.apollo.kmpsample.type.RepositoryOrderField
+import kotlinx.coroutines.invoke
+
+/**
+ * An implementation of a [GitHubDataSource] that shows how we can use coroutines to make our apollo requests.
+ */
+class ApolloCoroutinesService(private val apolloClient: ApolloClient) {
```

Comment:
It would be nice to move the ApolloCoroutinesService class to the multiplatform lib module since it is common logic between android/ios.

Reply:
For sure. I kind of like to do that in a separate PR. That would allow us to remove networking call from `SceneDelegate`. That is also misplaced atm.